pkg/azure/utils.go

```diff
@@ -577,11 +576,9 @@ func (d *MachinePlugin) deleteVMNicDisks(ctx context.Context, clients spi.AzureD
 
 	deleters := []func() error{nicDeleter, diskDeleter}
 
-	if dataDiskNames != nil {
```

Comment:
Why is this check removed? I hope this won't cause issues.

Reply:
I think it's okay - https://github.com/gardener/machine-controller-manager-provider-azure/blob/master/pkg/azure/core.go#L144. Looks like it's never nil.

Reply:
If it is nil then the loop is never executed. No panics occur. So, this is good, @prashanth26 😄